x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -135,23 +135,35 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
-  public async rotateKeyPair(): Promise<boolean> {
-    const isRemoved = await this.removeKeyPair();
-    if (isRemoved) {
+  public async rotateKeyPair(): Promise<{ error: Error } | undefined> {
```

Comment:
IMO, this should throw in case of error and not returning an object with error field. `Promise<void>`

Reply:
done c1976b5a98fec83b21f3218c46ebd8f6f3869340